intermediate-python-testing.md

```diff
@@ -363,6 +363,29 @@ mock_handle = mock.patch('path.to.CLASS_YOU_ARE_MOCKING')
 mock_handle.return_value = mocked_class
 ```
 
+#### `patch()` ####
+
+`patch()` is the main method that allows you to replace a dependency with a mock. There are two main ways to use this method -- as a decorator or as a context manager.
+
+As a decorator, `@patch()` can be used on a testing class or a single test function. The mock that it creates will live for the duration of the class or function. Once the test (or all the tests in the class, as the case may be) is complete, the mock is cleaned up and any other tests will use the real dependency instead of the mock.
+
+As a context manager, `patch()` allows your mock to be scoped to just part of a function.
+
+##### Useful mock object attributes #####
+
+Part of a mock object's magic is that it does not necessarily behave like the object it's patching. For example, any attribute you call on your mock pops into existence by virtue of being called, even if it doesn't exist on the real dependency. While this can be helpful at times, mock objects can be more useful if they throw errors, return values, and otherwise behave like the objects, functions, or other dependencies they temporarily replace. 
+
+Both the `patch()` method and the mock object itself have attributes that allow you to more closely mimic the behavior of the dependency you want to temporarily replace.
```

Comment:
A note on attributes vs. arguments: 

`spec`, `autospec`, and `spec_set` are arguments – that is, values passed to the `patch()` method – that determine attributes of the mock instance, but aren't stored as attributes of the mock.

`return_value` and `side_effect` are arguments, too, but they're also attributes of the initialized mock – that is, you can access them from your mock instance. 

Does that make sense?